farm/data_handler/processor.py

```diff
@@ -1227,15 +1237,21 @@ def dataset_from_dicts(self, dicts, indices=None, return_baskets=False):
         self._init_samples_in_baskets()
         self._featurize_samples()
         if 0 in indices:
-            self._log_samples(2)
+            self._log_samples(1)
         # This mode is for inference where we need to keep baskets
         if return_baskets:
             dataset, tensor_names = self._create_dataset(keep_baskets=True)
-            return dataset, tensor_names, self.baskets
+            ret = [dataset, tensor_names]
```

Comment:
I am not sure about the consequences, but lists are mutable and tuples (the return type before) are not. This might introduce some pretty weird behaviour. Do we need it?

Please also check the other return statement above

Reply:
I wrote it like this to avoid having too many nested if statements since both `return_baskets` and `return_problematic` can alter the number of objects that are returned. Converting `ret` into a tuple in the next commit to avoid this mutability.